addOns/exportreport/src/main/java/org/zaproxy/zap/extension/exportreport/export/ReportExportPDF.java

```diff
@@ -0,0 +1,1111 @@
+                        extensionExport
+                                .getMessages()
+                                .getString("exportreport.details.requestbody.label"))
+                || detailsToInclude.contains(
+                        extensionExport
+                                .getMessages()
+                                .getString("exportreport.details.responseheader.label"))
+                || detailsToInclude.contains(
+                        extensionExport
+                                .getMessages()
+                                .getString("exportreport.details.responsebody.label"))) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Joins similar alerts to reduce output size
+     *
+     * @param extension
+     * @return
+     */
+    public java.util.List<List<Alert>> joinSimilarAlerts(ExtensionExportReport extension) {
+        // Make list of all alerts, joining same alerts
+        java.util.List<List<Alert>> alerts = new ArrayList<>();
+        java.util.List<Alert> allAlerts = extension.getAllAlerts();
+        Collections.sort(allAlerts, Collections.reverseOrder());
+
+        // join same alerts
+        for (int i = 0; i < allAlerts.size(); i++) {
+            Alert alertAllAlerts = allAlerts.get(i);
+            alerts.add(extension.getAlertsSelected(alertAllAlerts));
+            for (int j = 0; j < allAlerts.size(); j++) {
+                Alert alertToCompare = allAlerts.get(j);
+                if (alertAllAlerts.getName().equals(alertToCompare.getName())) {
+                    allAlerts.remove(j);
+                    j = 0;
+                }
+            }
+            i = 0;
+        }
+
+        return alerts;
+    }
+
+    /**
+     * Removes all alerts which have a risk level not to be included
+     *
+     * @param extension
+     * @param alerts
+     * @return
+     */
+    public java.util.List<List<Alert>> filterAlertsByRiskLevel(
+            ExtensionExportReport extension, List<List<Alert>> alerts) {
+        /*
+         * Output list modifications:
+         *  - Remove alerts that are not to be included based on risk names
+         *  - If the alert is included, remove details that are not to be included
+         */
+        ArrayList<String> selectedRisks = extension.getIncludedAlertSeverity();
+        ArrayList<String> alertRisks = extension.getAlertSeverity();
+        boolean isSelected;
+
+        for (int i = alerts.size() - 1; i >= 0; i--) {
+            List<Alert> alertList = alerts.get(i);
+            isSelected = false;
+
+            for (int j = alertList.size() - 1; j >= 0; j--) {
+                Alert alert = alertList.get(j);
+
+                for (int k = 0; k < selectedRisks.size(); k++) {
+                    // If the alert risk level is in in the include list
+                    if (alertRisks.get(alert.getRisk()).equals(selectedRisks.get(k))) {
```

Comment:
This (`alertRisks.get(alert.getRisk())`) should not rely on the order shown in the UI (e.g. could do the same as in `ReportExport` with `mapRiskToTranslation`).

Reply:
From my understanding of the original implementation in `alertreport`, the idea was to check all alerts in the list and remove the ones for which the risk level was not included by the user from the UI.
I think the entire loop (for `k`) can be replaced with your previous comment (using `selectedRisks.contain` instead of the loop and use of `isSelected`), bypassing the need for mapping.

Reply:
That should work as long as the values were/are already mapped.